cell_based/src/cell_based_pde/ParabolicBoxDomainPdeModifier.cpp

```diff
@@ -161,6 +184,166 @@ void ParabolicBoxDomainPdeModifier<DIM>::SetupInitialSolutionVector(AbstractCell
+        if (max_radius < radius)
+        {
+            max_radius = radius;
+        }                
+    }
+
+    // Create mesh from cell centres so can interpolate tissue velocity onto mpFeMesh
+    std::vector<Node<DIM>*> temp_nodes;
+    std::vector<c_vector<double,DIM>> cell_displacements;
+    unsigned cell_index = 0;
+    for (typename AbstractCellPopulation<DIM>::Iterator cell_iter = rCellPopulation.Begin();
+        cell_iter != rCellPopulation.End();
+        ++cell_iter)
+    {
+        c_vector<double,DIM> position_of_cell = rCellPopulation.GetLocationOfCellCentre(*cell_iter);
+        cell_index++;
+
+        // Only use cells that were in both this and the previous timestep.
+        if (mOldCellLocations.find(*cell_iter) != mOldCellLocations.end())
+        {
+            //PRINT_VARIABLE(mOldCellLocations[*cell_iter]);
+            c_vector<double,DIM> displacement = position_of_cell - mOldCellLocations[*cell_iter];
+       
+            cell_displacements.push_back(displacement);
+            temp_nodes.push_back(new Node<DIM>(cell_index, position_of_cell));
+        }
+    }
+    MutableMesh<DIM,DIM> cell_mesh(temp_nodes);
+
+    // Make the deformed mesh. Based on the displacement of cells. 
+    TetrahedralMesh<DIM, DIM>* p_deformed_mesh = new TetrahedralMesh<DIM,DIM>();
+    this->GenerateAndReturnFeMesh(this->mpMeshCuboid,this->mStepSize,p_deformed_mesh);
+    
+    for (typename TetrahedralMesh<DIM, DIM>::NodeIterator node_iter = p_deformed_mesh->GetNodeIteratorBegin();
+         node_iter != p_deformed_mesh->GetNodeIteratorEnd();
+         ++node_iter)
+    {
+        c_vector<double, DIM> node_location = node_iter->rGetLocation();
+
+        
+        c_vector<double, DIM> new_node_location = node_location;
+        
+        if (norm_2(node_location) <= max_radius)
+        {
+            // Find the element in the cell mesh that contains this node.
+            try
+            {
+                unsigned elem_index = cell_mesh.GetContainingElementIndex(node_location, false);
+        
+                // Now do the interpolation
+                Element<DIM,DIM>* p_element = cell_mesh.GetElement(elem_index);
+                c_vector<double,DIM+1> weights = p_element->CalculateInterpolationWeights(node_location);
+
+                c_vector<double,DIM> interpolated_cell_displacement = zero_vector<double>(DIM);
+                for (unsigned i=0; i<DIM+1; i++)
+                {
+                    c_vector<double,DIM> nodal_value = cell_displacements[p_element->GetNodeGlobalIndex(i)];
+                    interpolated_cell_displacement += nodal_value * weights(i);
+                }
+                new_node_location = node_location + interpolated_cell_displacement;
+                node_iter->rGetModifiableLocation() = new_node_location;
+            }
+            catch (Exception&) // not_in_mesh
+            {
+                //Don't do anything as these FE nodes are outside of the Cell mesh.               
```

Comment:
Does this want a warning or anything? Or at least a check that the Exception message is the intended one?

Reply:
It does will add one and test it's thrown. OMG this message appeared in VScode ite like its 2024!!